remove ref to testCondDBDict which is not part of this package any more

smuzaffar opened the pull request.

smuzaffar committed:
remove ref to testCondDBDict which is not part of this package any more

smuzaffar commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by @smuzaffar (Malik Shahzad Muzaffar) for CMSSW_7_6_X.

remove ref to testCondDBDict which is not part of this package any more

It involves the following packages:

CondCore/CondDB


@ggovi, @cmsbuild can you please review it and eventually sign? Thanks.
@ghellwig, @apfeiffer1 this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

davidlt commented:
I commented on this Saturday. _testRootStreaming_ needs the dictionary, otherwise it will fail. The dictionary files and selections files are kept, but for some reason building dictionary was disabled. (See #11120).

It was removed to silence some warning from SCRAM: https://github.com/apfeiffer1/cmssw/commit/c7b8f3e62d0e1a78de336ddc77f84e6012e3369a#diff-156022378736cc699d979c4a9edf1664

smuzaffar commented:
@davidlt, ah ok, yes testRootStreaming need the MyTestData dictionary in order to run. I think just added a empty cc file and generating a library in this test directory should allow to generate the missing dictionary

davidlt commented:
There is all the files need for _genreflex_ command. That generates a C++ file for dictionary, aren't that enough?

smuzaffar committed:
generate test dictionary

cmsbuild commented:
Pull request #11628 was updated. @ggovi, @cmsbuild can you please check and sign again.

smuzaffar commented:
yes but one needs to explicitly tell scram that it should also generate a library along with other products on the test directory. Does does it automatically for src sub-directory but of all the other sub-dirs one needs to explicitly ask for build products. I have updated the PR where I have just updated the test/BuildFile.xml  so that it can generate a share library (magic part is that here it doe snot need a cc file)

smuzaffar commented:
please test

davidlt commented on CondCore/CondDB/test/BuildFile.xml:
```diff
@@ -17,11 +18,9 @@
 </bin>
 
 <bin   file="testRootStreaming.cpp" name="testRootStreaming">
-  <lib   name="testCondDBDict"/>
```
Does SCRAM automatically link the dictionary to everything, or we need to bring back this statement?

smuzaffar replied:
I think root is going to take care of it by reading the testCondDBDict_xr.rootmap file ...right? at least unit tests runs locally without explicitly linking to testCondDBDict

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c919e61ba925b25933c7ce3aa465e0ef122365a0
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11628/8609/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11628/8609/summary.html

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_7_6_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @degano, @smuzaffar

davidlange6 commented:
+1
